src/content_script/mastodonInject.js

```diff
@@ -137,8 +152,12 @@ async function injectInteractionButtons() {
  * @returns {void}
  */
 function initInjections() {
-    injectFollowButton().catch(console.error);
-    injectInteractionButtons().catch(console.error);
+    checkLoggedIn().then((r) => {
+        if (!r) {
+            injectFollowButton().catch(console.error);
+            injectInteractionButtons().catch(console.error);
+        }
+    });
```

Comment:
```suggestion
    checkLoggedIn().then((isLoggedIn) => {
        if (isLoggedIn) {
            return;
        }
        
        injectFollowButton().catch(console.error);
        injectInteractionButtons().catch(console.error);
    });
```

early return + more clean code IMHO.

Or please actually even better with await:
```suggestion
    const isLoggedIn = await checkLoggedIn();
    if (isLoggedIn) {
        return;
    }
        
        
    injectFollowButton().catch(console.error);
    injectInteractionButtons().catch(console.error);
```

+ make the function async

Reply:
Making it async/await throws a ReferenceError (accessing before initialization) so the first choice is taken.

Reply:
> Making it async/await throws a ReferenceError (accessing before initialization)

Uff, why? Actually that should just be a different syntax for the same result.

Reply:
```js
async function initInjections() {
    const isLoggedIn = await isLoggedIn();
    if (isLoggedIn) {
        return;
    }
    injectFollowButton().catch(console.error);
    injectInteractionButtons().catch(console.error);
}
```

<img width="522" alt="Screenshot 2023-03-19 at 10 57 35" src="https://user-images.githubusercontent.com/16656689/226184398-234b08dd-635b-4024-b6b4-9ccd7a7a836c.png">

Reply:
crazy…